lib/ClangImporter/ClangImporter.cpp

```diff
@@ -8529,7 +8531,23 @@ static bool hasUnsafeType(Evaluator &evaluator, clang::QualType clangType) {
     // All other pointers are considered unsafe.
     return true;
   }
-  
+
+  // Consider functions without prototype as unsafe.
+  if (clangType->getAs<clang::FunctionNoProtoType>())
+    return true;
+
+  if (auto fnTy = clangType->getAs<clang::FunctionProtoType>()) {
+    if (hasUnsafeType(evaluator, fnTy->getReturnType()))
+      return true;
+    for (auto paramTy : fnTy->getParamTypes()) {
+      // Do not consider reference parameters as unsafe by default.
+      if (paramTy->isReferenceType())
+        paramTy = paramTy->getPointeeType();
```

Comment:
Would this work correctly for foreign reference types? Could you please add a test?